addons/disposable/fnc_firedDisposable.sqf

```diff
@@ -0,0 +1,101 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_firedDisposable
+
+Description:
+    Handles firing a disposable weapon.
+
+Parameters:
+    _unit       - Unit that fired the disposable weapon <OBEJCT>
+    _launcher   - Disposable weapon <STRING>
+    _muzzle     - Muzzle fired by the disposable weapon <STRING>
+    _mode       - Current weapon mode of the disposable weapon <STRING>
+    _ammo       - Ammo fired by the disposable weapon <STRING>
+    _magazine   - Current magazine of the disposable weapon <STRING>
+    _projectile - Fired projectile <OBEJCT>
+    _unit       - Always same as element 0 <OBEJCT>
+
+Returns:
+    Nothing.
+
+Examples:
+    (begin example)
+        class EventHandlers {
+            fired = "_this call CBA_fnc_firedDisposable";
+        };
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+
+params ["_unit", "_launcher", "_muzzle", "", "", "", "_projectile"];
```

Comment:
can be done in a later release, but we should add some way of handling spotting rounds (LAW 80)

Reply:
Yeah, this is why I don't like dropping the launcher. It needs `addWeaponWithAttachmentsCargoGlobal` to work.

Reply:
maybe
```
_wh = "GroundWeaponHolder_Scripted" createVehicle position _unit;
_unit action ["DropWeapon", _wh, _usedLauncher];
```

Reply:
DropWeapon is broken. Always has been always will be.

Reply:
`addWeaponWithAttachmentsCargo` will be fully implemented in 1.96 (contact DLC is 1.94)